sos/plugins/gluster_block.py

```diff
@@ -0,0 +1,45 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+import glob
+from sos.plugins import Plugin, RedHatPlugin
+
+
+class GlusterBlock(Plugin, RedHatPlugin):
+    """Gluster Block"""
+
+    plugin_name = 'gluster_block'
+    profiles = ('storage',)
+    packages = ["gluster-block"]
+    files = ["/usr/sbin/gluster-block"]
+    config_to_collect = ["/sys/kernel/config/target", "/etc/target"]
+
+    def setup(self):
+        self.add_cmd_output("targetcli ls")
+
+        # collect config files
+        for config_file in self.config_to_collect:
+            self.add_copy_spec(config_file)
+
+        # collect logs - apply log_size for any individual file
+        # all_logs takes precedence over logsize
+        if not self.get_option("all_logs"):
+            limit = self.get_option("log_size")
+        else:
+            limit = 0
+
+        if limit:
+            for f in glob.glob("/var/log/gluster-block/*.log"):
```

Comment:
Ack. This is ugly, but it is in keeping with other plugins that try to work around the current limits API shortcomings. We can clean it up later when those are addressed.